src/action.py

```diff
@@ -28,6 +29,26 @@ def _absolute_url(url: types_.Url | None, discourse: Discourse) -> types_.Url |
     return discourse.absolute_url(url=url) if url is not None else None
 
 
+def _log_content_change(old: str, new: str) -> None:
+    """Log the difference between the old and new content, if any.
+
+    Args:
+        old: The previous content.
+        new: The current content.
+    """
+    old = f"{old}\n" if not old.endswith("\n") else old
+    new = f"{new}\n" if not new.endswith("\n") else new
+    if new != old:
+        logging.info(
+            "content change:\n%s",
```

Comment:
The name of changed files should be added to the log message, to make it more clear.

Reply:
This appears right after the log message with the action (at the start of this function) which includes the file being changed. Do you think that is sufficiently clear?

Reply:
This line logs the action: `logging.info("dry run: %s, action: %s", dry_run, action)` which has the following attributes:

```Python
@dataclasses.dataclass
class UpdateAction:
    """Represents a page to be updated.

    Attrs:
        level: The number of parents, is 1 if there is no parent.
        path: The a unique string identifying the navigation table row.
        navlink_change: The changeto the navlink.
        content_change: The change to the documentation content.
    """

    level: Level
    path: TablePath
    navlink_change: NavlinkChange
    content_change: ContentChange | None
```